birdy/ipyleafletwfs/examples/ipyleafletwfs_guide.ipynb

```diff
@@ -0,0 +1,342 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### A search by ID for features is also available. Let's set back the main widget to default so we can have access to feature IDs again"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "display_data",
+     "data": {
+      "text/plain": "Map(center=[46.42, -64.14], controls=(ZoomControl(options=['position', 'zoom_in_text', 'zoom_in_title', 'zoom_…",
+      "application/vnd.jupyter.widget-view+json": {
+       "version_major": 2,
+       "version_minor": 0,
+       "model_id": "b8b9e2caf58b4feb8ba6cec7fd36a767"
+      }
+     },
+     "metadata": {}
+    }
+   ],
+   "source": [
+    "wfs_connection.create_feature_property_widget(widget_name='main_widget')\n",
+    "demo_map"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Now click on a feature and replace '748' in the cell below with a new ID number to get the full properties of that feature"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "execute_result",
+     "data": {
+      "text/plain": "{'Hylak_id': 748,\n 'Lake_name': 'Grand',\n 'Country': 'Canada',\n 'Continent': 'North America',\n 'Poly_src': 'CanVec',\n 'Lake_type': 1,\n 'Grand_id': 0,\n 'Lake_area': 173.33,\n 'Shore_len': 229.49,\n 'Shore_dev': 4.92,\n 'Vol_total': 2042.27,\n 'Vol_res': 0,\n 'Vol_src': 3,\n 'Depth_avg': 11.8,\n 'Dis_avg': 81.913,\n 'Res_time': 288.6,\n 'Elevation': 0,\n 'Slope_100': 1.97,\n 'Wshd_area': 3792.1,\n 'Pour_long': -66.12332,\n 'Pour_lat': 45.839487,\n 'bbox': [-66.2071, 45.8363, -65.8713, 46.1022]}"
+     },
+     "metadata": {},
+     "execution_count": 10
+    }
+   ],
+   "source": [
+    "wfs_connection.feature_properties_by_id(748)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### To get rid of all the property widgets:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "display_data",
+     "data": {
+      "text/plain": "Map(center=[46.42, -64.14], controls=(ZoomControl(options=['position', 'zoom_in_text', 'zoom_in_title', 'zoom_…",
+      "application/vnd.jupyter.widget-view+json": {
+       "version_major": 2,
+       "version_minor": 0,
+       "model_id": "b8b9e2caf58b4feb8ba6cec7fd36a767"
+      }
+     },
+     "metadata": {}
+    }
+   ],
+   "source": [
+    "wfs_connection.clear_property_widgets()\n",
+    "demo_map"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "### And finally, to remove the layer from the map"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wfs_connection.remove_layer()"
```

Comment:
@huard  asked me to take a look at this as well ... Have started here runnning on a local env and everything ran up to the `remove_layer()` call. Traceback :

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-15-4d3049f72d4a> in <module>
----> 1 wfs_connection.remove_layer()

~/github/github_birdy/birdy/ipyleafletwfs/base.py in remove_layer(self)
    224         if self._layer:
    225             # Remove maps elements
--> 226             self.clear_property_widgets()
    227             self._source_map.remove_control(self._refresh_widget)
    228             self._source_map.remove_layer(self._layer)

~/github/github_birdy/birdy/ipyleafletwfs/base.py in clear_property_widgets(self)
    341           The map instance from which the widgets are to be removed.
    342         """
--> 343         for widget in self._property_widgets:
    344             self._source_map.remove_control(self._property_widgets[widget]['widget'])
    345         self._property_widgets = None

TypeError: 'NoneType' object is not iterable
```

Reply:
Thanks for that catch. Forgot to change this after another change. 

Adding a conditional check to clear_property_widgets() to prevent this